src/zenml/integrations/label_studio/annotators/label_studio_annotator.py

```diff
@@ -0,0 +1,661 @@
+
+    def get_converted_dataset(
+        self, dataset_id: int, output_format: str
+    ) -> Dict[Any, Any]:
+        """Extract annotated tasks in a specific converted format.
+
+        Args:
+            dataset_id: Id of the dataset.
+            output_format: Output format.
+
+        Returns:
+            A dictionary containing the converted dataset.
+        """
+        # project = self._dataset_name_to_project(dataset_name)
+        project = self.get_dataset(dataset_id)
+        return project.export_tasks(export_type=output_format)
+
+    def get_labeled_data(
+        self, *args: Any, dataset_name: str, **kwargs: Any
+    ) -> Any:
+        """Gets the labeled data for the given dataset.
+
+        Args:
+            dataset_name: Name of the dataset.
+            *args: Additional arguments to pass to the Label Studio client.
+            **kwargs: Additional keyword arguments to pass to the Label Studio client.
+
+        Returns:
+            A dictionary containing the labeled data.
+        """
+        ls = self._get_client()
+        dataset_id = self.get_id_from_name(dataset_name)
+        return ls.get_project(dataset_id).get_labeled_tasks()
+
+    def get_unlabeled_data(self, **kwargs: str) -> Any:
+        """Gets the unlabeled data for the given dataset.
+
+        Args:
+            **kwargs: Additional keyword arguments to pass to the Label Studio client.
+
+        Returns:
+            A dictionary containing the unlabeled data.
+        """
+        ls = self._get_client()
+        dataset_name = kwargs.get("dataset_name")
+        dataset_id = self.get_id_from_name(dataset_name)
+        return ls.get_project(dataset_id).get_unlabeled_tasks()
+
+    def register_dataset_for_annotation(
+        self,
+        config: LabelStudioDatasetRegistrationConfig,
+    ) -> Any:
+        """Registers a dataset for annotation.
+
+        Args:
+            config: Configuration for the dataset.
+
+        Returns:
+            A Label Studio Project object.
+        """
+        ls = self._get_client()
+
+        if self.get_id_from_name(config.dataset_name):
+            dataset = ls.get_project(self.get_id_from_name(config.dataset_name))
+        else:
+            dataset = self.add_dataset(
+                name=config.dataset_name,
+                label_config=config.label_config,
+            )
+
+        return dataset
+
+    def _get_azure_import_storage_sources(
+        self, dataset_id: int
+    ) -> List[Dict[str, Any]]:
+        """Gets a list of all Azure import storage sources.
+
+        Args:
+            dataset_id: Id of the dataset.
+
+        Returns:
+            A list of Azure import storage sources.
+
+        Raises:
+            ConnectionError: If the connection to the Label Studio backend is unavailable.
+        """
+        # TODO: check if client actually is connected etc
+        ls = self._get_client()
+        query_url = f"/api/storages/azure?project={dataset_id}"
```

Comment:
This seems to be identical with the two methods below apart from the `azure/gcs/s3` part of the URL? We could probably combine those right?

Reply:
What do you mean combine them? Those methods are implemented by Label Studio. Do you mean we could implement a wrapper around them? We'd still have the conditional living somewhere...

Reply:
Yes exactly, what I meant here is:
`_get_azure_import_storage_sources`, `_get_gcs_import_storage_sources` and `_get_s3_import_storage_sources` are identical, apart from one tiny part of the URL.

We could simply have a method like
```python
def _get_import_storage_sources(..., provider: str):
    ls = self._get_client()
    query_url = f"/api/storages/{provider}?project={dataset_id}"
    ...
```

which removes the duplicated code. Probably `provider` is the wrong name here, but I hope you get the idea)